src/hooks/useCart/useCart.tsx

```diff
@@ -235,11 +236,29 @@ export const useCart = (
    */
   const _completeCheckout = useCallback(() => {
     const complete = async (): Promise<void> => {
-      const transactions = Object.values(cart)
-        .filter(({ quantity }) => quantity)
-        .map(({ category, quantity, identifierInputs }) => {
-          return { category, quantity, identifierInputs };
-        });
+      const allCleanedIdentifierInputs: IdentifierInput[] = [];
+      let transactions: any = [];
+      try {
+        transactions = Object.values(cart)
+          .filter(({ quantity }) => quantity)
+          .map(({ category, quantity, identifierInputs }) => {
+            const cleanedIdentifierInputs = cleanIdentifierInputs(
+              identifierInputs
+            );
+            allCleanedIdentifierInputs.push(...cleanedIdentifierInputs);
+            return {
+              category,
+              quantity,
+              identifierInputs: cleanedIdentifierInputs,
+            };
+          });
+        validateIdentifierInputs(allCleanedIdentifierInputs);
+      } catch (error) {
+        setCartState("DEFAULT");
```

Comment:
do add the check for empty transactions here too, and subsequently throw the error message `new Error(ERROR_MESSAGE.MISSING_SELECTION))`

Reply:
done!